File: nodejs/src/entities/PersonaAtributo.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from 'typeorm';
import { Persona } from './Persona';
import { AtributoTipo } from './AtributoTipo';

@Entity('persona_atributos')
export class PersonaAtributo {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  personaId: number;

  @Column()
  atributoTipoId: number;

  @Column({ type: 'varchar', length: 500 })
  valor: string;

  @CreateDateColumn()
  fechaCreacion: Date;

  @UpdateDateColumn({ nullable: true })
  fechaModificacion?: Date;

  @ManyToOne(() => Persona, persona => persona.personaAtributos, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'personaId' })
  persona: Persona;

  @ManyToOne(() => AtributoTipo, atributoTipo => atributoTipo.personaAtributos, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'atributoTipoId' })
  atributoTipo: AtributoTipo;
}

File: dotnet/PersonasABM.Domain/Entities/PersonaAtributo.cs
using System.ComponentModel.DataAnnotations;

namespace PersonasABM.Domain.Entities;

public class PersonaAtributo
{
    public int Id { get; set; }
    
    public int PersonaId { get; set; }
    
    public int AtributoTipoId { get; set; }
    
    [Required]
    [StringLength(500)]
    public string Valor { get; set; } = string.Empty;
    
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    
    public DateTime? FechaModificacion { get; set; }
    
    // Navegación
    public virtual Persona Persona { get; set; } = null!;
    public virtual AtributoTipo AtributoTipo { get; set; } = null!;
}
